WordPress/gutenberg
# Try adding a close icon for navigation sidebar

## Description
This PR implements [the design by @jasmussen to **add a close icon in the Navigation Sidebar component**](https://github.com/WordPress/gutenberg/issues/29529#issuecomment-791250758). Closes https://github.com/WordPress/gutenberg/issues/29529.

This is to solve the issue (reported many times) where folks find it hard to discover that the `W` (WordPress logo) acts as the toggle for open/closing the sidebar.

Currently, this **works only within the Site Editor experiment**. If acceptable it will need to be mirrored in the Post Editor implementation.



## How has this been tested?

* Activate TT1 Theme.
* Go to Site Editor.
* Toggle the Navigation Sidebar open.
* See a new `X` icon appear on the RHS of the panel that opens.
* Click `X` to close it.
* Also note that the WordPress logo `W` still acts as a toggle. 

## Screenshots 

https://user-images.githubusercontent.com/444434/115403289-2e84c280-a1e4-11eb-98eb-9a92de434363.mov







## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
